### PR TITLE
Add type annotations to core classes

### PR DESCRIPTION
**Pull request by davidmreed**
This PR adds type annotations for several core classes. There are a couple of small code changes to serve this objective, but there's intended to be no change in behavior.

### CODE REVIEW

File: cumulusci/core/flowrunner.py [hunk 1]
```diff
@@ -140,13 +160,32 @@ def __repr__(self):
...
+        result: Any,
+        return_values: Any,
+        exception: Optional[Exception],
+    ):
+        self.step_num = step_num
+        self.task_name = task_name
+        self.path = path
+        self.result = result
+        self.return_values = return_values
+        self.exception = exception
```
Comment (jstvz) [inline cumulusci/core/flowrunner.py:185]: ↔️: [`typing.NamedTuple`](https://docs.python.org/3.8/library/typing.html#typing.NamedTuple)
 ```python
class StepResult(typing.NamedTuple):
    step_num: StepVersion
    task_name: str
    path: str
    result: Any
    return_values: Any
    exception: Optional[Exception]
```

File: cumulusci/core/flowrunner.py [hunk 2]
```diff
@@ -51,26 +51,34 @@
...
 from operator import attrgetter
+from typing import (
+    Any,
+    DefaultDict,
+    Dict,
+    Optional,
+    Tuple,
+    Type,
+    TYPE_CHECKING,
+    List,
```
Comment (jstvz) [inline cumulusci/core/flowrunner.py:67]: Suggestions:
- `pre-commit install --install-hooks`
- `git commit --no-verify -m 'wip'` if you get annoyed.
- Also: https://marketplace.visualstudio.com/items?itemName=ms-python.isort

**Review by jstvz — COMMENTED** [review]
Nice.

File: cumulusci/core/flowrunner.py (hunk 1 above)
Comment (davidmreed) [reply cumulusci/core/flowrunner.py:185]: TIL. That's clean. Will use that approach to avoid adding code.

File: cumulusci/core/runtime.py [hunk 3]
```diff
@@ -38,30 +38,30 @@ def __init__(self, *args, load_keychain=True, **kwargs):
...
     @property
-    def universal_config_cls(self):
+    def universal_config_cls(self) -> Type:
         klass = self.get_universal_config_class()
         return klass or self.universal_config_class
 
     @abstractmethod
-    def get_universal_config_class(self):
+    def get_universal_config_class(self) -> Optional[Type]:
         return None
```
Comment (jstvz) [inline cumulusci/core/runtime.py:47]: I think `pass` or `...` works with `Type` here ([but not](https://github.com/python/mypy/blob/master/docs/source/class_basics.rst#abstract-base-classes-and-multiple-inheritance) with `super().get_universal...`) but no objection if we come back to it later

**Review by jstvz — APPROVED** [review]
🧗🏾

**Pull request merged**